x-pack/heartbeat/monitors/browser/source/zipurl_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package source
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/v7/x-pack/heartbeat/monitors/browser/source/fixtures"
+)
+
+func TestZipUrlFetchNoAuth(t *testing.T) {
+	address, teardown := setupTests()
```

Comment:
I thought doing this for every test was a overkill which is why i moved up everything in to functions and called them from a global test function. I guess it's not?  Just a question though.